htdocs/core/modules/commande/doc/doc_generic_order_odt.modules.php

```diff
@@ -305,8 +305,10 @@ public function write_file($object, $outputlangs, $srctemplatepath, $hidedetails
 				$contactobject = null;
 				if (!empty($usecontact))
 				{
-					if ($usecontact && ($object->contact->fk_soc != $object->thirdparty->id && (!isset($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT) || !empty($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT)))) {
-						$socobject = $object->contact;
+					if ($usecontact && ($object->contact->fk_soc != $object->thirdparty->id && (!empty($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT)))) {
+						$object->contact->fetch_thirdparty();
```

Comment:
Goal of option MAIN_USE_COMPANY_NAME_OF_CONTACT is to replace the company information with the contact.
So we must keep 
$socobject = $object->contact;
The object of company is replaced with the object of contact.

The behaviour you introduce is already managed by the "else".

Reply:
I fully understand the purpose of MAIN_USE_COMPANY_NAME_OF_CONTACT but the current implementation just does not work, because later the function function get_substitutionarray_thirdparty is called with this object and this function expects a thirdparty object (and not a contact object). I've tested it and with my implementation it works in both cases (MAIN_USE_COMPANY_NAME_OF_CONTACT is set or not set)

Reply:
Sorry i gave you bad information. Test should be
`if ($usecontact && ($object->contact->fk_soc != $object->thirdparty->id && (!isset($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT) && !empty($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT)))) {`
So like it is in old version.
What we want is to have MAIN_USE_COMPANY_NAME_OF_CONTACT the default value.
so if unset, we must act as if it is set to MAIN_USE_COMPANY_NAME_OF_CONTACT=1
Only case MAIN_USE_COMPANY_NAME_OF_CONTACT set to 0 will switch in the else.
Really Sorry. Can you fix my error too ?

Reply:
Still trouble with the test 
`if ($usecontact && ($object->contact->fk_soc != $object->thirdparty->id && (isset($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT) && !empty($conf->global->MAIN_USE_COMPANY_NAME_OF_CONTACT)))) {`
Old code looks good.